From 8b0956286967e92bd68c04eb4e051225f00fce44 Mon Sep 17 00:00:00 2001
From: Jiri Moskovcak <jmoskovc@redhat.com>
Date: Thu, 12 Jul 2012 11:24:39 +0200
Subject: [PATCH 070/105] oops: don't create oops dir in reverse rhbz#814594

when kernel starts printing oops, only very first one is not tainted.
If we start in reverse, it will firstly create direcotry with tainted oops
and others will be deleted as dup of first one (if there are same), thus
we loose NOT tainted oops.

Signed-off-by: Nikola Pajkovsky <npajkovs@redhat.com>
---
 src/plugins/abrt-dump-oops.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/plugins/abrt-dump-oops.c b/src/plugins/abrt-dump-oops.c
index 48d7b91..849ca4e 100644
--- a/src/plugins/abrt-dump-oops.c
+++ b/src/plugins/abrt-dump-oops.c
@@ -111,9 +111,9 @@ static int scan_syslog_file(GList **oops_list, int fd, struct stat *statbuf, int
 static unsigned save_oops_to_dump_dir(GList *oops_list, unsigned oops_cnt)
 {
     unsigned countdown = 16; /* do not report hundreds of oopses */
-    unsigned idx = oops_cnt;
+    unsigned idx = 0; //oops_cnt;
 
-    VERB1 log("Saving %u oopses as dump dirs", idx >= countdown ? countdown-1 : idx);
+    VERB1 log("Saving %u oopses as dump dirs", oops_cnt >= countdown ? countdown-1 : oops_cnt);
 
     char *cmdline_str = NULL;
     FILE *cmdline_fp = fopen("/proc/cmdline", "r");
@@ -137,9 +137,9 @@ static unsigned save_oops_to_dump_dir(GList *oops_list, unsigned oops_cnt)
 
     pid_t my_pid = getpid();
     unsigned errors = 0;
-    while (idx != 0 && --countdown != 0)
+    while (idx <= oops_cnt && --countdown != 0)
     {
-        char *first_line = (char*)g_list_nth_data(oops_list, --idx);
+        char *first_line = (char*)g_list_nth_data(oops_list, idx++);
         char *second_line = (char*)strchr(first_line, '\n'); /* never NULL */
         *second_line++ = '\0';
 
-- 
1.7.11.2

